tests/mav_frame_tests.rs

```diff
@@ -0,0 +1,97 @@
+pub mod test_shared;
+
+mod mav_frame_tests {
+    use mavlink::common::MavMessage;
+    use mavlink::MavFrame;
+    use mavlink::MavHeader;
+
+    // NOTE: No header, based over get_heartbeat_msg
+    pub const HEARTBEAT_V2: &[u8] = &[
+        crate::test_shared::COMMON_MSG_HEADER.sequence,
+        crate::test_shared::COMMON_MSG_HEADER.system_id,
+        crate::test_shared::COMMON_MSG_HEADER.component_id,
+        0x00, // msg ID
+        0x00,
+        0x00,
+        0x05, // payload - custom_mode
+        0x00, //
+        0x00, //
+        0x00, //
+        0x02, // mav_type
+        0x03, // autopilot
+        0x59, // base_mode
+        0x03, // system_status
+        0x03, // mavlink_version
+        0x10, // checksum
+        0xf0,
+    ];
+
+    #[test]
+    pub fn test_deser_ser() {
+        use mavlink::{common::MavMessage, MavFrame, MavlinkVersion};
+        let frame = MavFrame::<MavMessage>::deser(MavlinkVersion::V2, HEARTBEAT_V2)
+            .expect("failed to parse message");
+
+        assert_eq!(frame.header, crate::test_shared::COMMON_MSG_HEADER);
+        let heartbeat_msg = crate::test_shared::get_heartbeat_msg();
+
+        let buffer = frame.ser();
+        assert_eq!(buffer[..buffer.len() - 2], HEARTBEAT_V2[..buffer.len() - 2]);
```

Comment:
we are.. skipping the checksum?

Reply:
> we are.. skipping the checksum? reasons?

we do not calculate the checksum on the serialization function

Reply:
Ok, but then shouldn't it be `assert_eq!(buffer, HEARTBEAT_V2[..buffer.len()]);` instead? Otherwise, we'll be skipping more than that.